controllers/sriovnetworkpoolconfig_controller.go

```diff
@@ -0,0 +1,189 @@
+		return reconcile.Result{}, err
+	}
+
+	// examine DeletionTimestamp to determine if object is under deletion
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !sriovnetworkv1.StringInArray(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers) {
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, sriovnetworkv1.POOLCONFIGFINALIZERNAME)
+			if err := r.Update(context.Background(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		if utils.ClusterType == utils.ClusterTypeOpenshift {
+			if err = r.syncOvsHardwareOffloadMachineConfigs(instance, false); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if sriovnetworkv1.StringInArray(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers) {
+			// our finalizer is present, so lets handle any external dependency
+			reqLogger.Info("delete SriovNetworkPoolConfig CR", "Namespace", instance.Namespace, "Name", instance.Name)
+			if utils.ClusterType == utils.ClusterTypeOpenshift {
+				if err = r.syncOvsHardwareOffloadMachineConfigs(instance, true); err != nil {
+					// if fail to delete the external dependency here, return with error
+					// so that it can be retried
+					return reconcile.Result{}, err
+				}
+			}
+			// remove our finalizer from the list and update it.
+			instance.ObjectMeta.Finalizers = sriovnetworkv1.RemoveString(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers)
+			if err := r.Update(context.Background(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *SriovNetworkPoolConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&sriovnetworkv1.SriovNetworkPoolConfig{}).
+		Complete(r)
+}
+
+func (r *SriovNetworkPoolConfigReconciler) syncOvsHardwareOffloadMachineConfigs(nc *sriovnetworkv1.SriovNetworkPoolConfig, deletion bool) error {
+	logger := r.Log.WithName("syncOvsHardwareOffloadMachineConfigs")
+
+	mcpName := nc.Spec.OvsHardwareOffloadConfig.Name
+	mcName := "00-" + mcpName + "-" + OVS_HWOL_MACHINE_CONFIG_NAME_SUFFIX
+
+	foundMC := &mcfgv1.MachineConfig{}
+	mcp := &mcfgv1.MachineConfigPool{}
+
+	if mcpName == "" {
+		logger.Info("MachineConfigPool name is not defined in SriovNetworkPoolConfig", nc.GetNamespace(), nc.GetName())
+		return nil
+	}
+
+	if mcpName == "master" {
+		logger.Info("Master nodes are selected by OvsHardwareOffloadConfig.Name, ignoring.")
+		return nil
+	}
+
+	err := r.Get(context.TODO(), types.NamespacedName{Name: mcpName}, mcp)
```

Comment:
The variable `mcp` is not used after getting? Is that only for checking the existence of the MCP?

Reply:
Yes, only for checking the existence of MCP.